### ISSUE TITLE
Scope is not created/destroyed using angularCompileRows

### ISSUE THREAD

**Issue by smhc** [ag-grid/ag-grid#1753]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Scope of cellEditor and cellRenderer is not destroyed when DOM element is removed. (using angularCompileRows)

**Expected behavior**

A new scope should be created for each edit/render so that it can be destroyed when the element is removed. Currently there is a single parent scope for the row which does not get destroyed. This approach only works if the templates do not create new scopes, but this is pretty rare.

If the template itself creates new scopes (isolate scopes, scopes via ng-if etc) they are not destroyed until the row scope is destroyed.  Removing the element when switching between edit/render is insufficient, the associated scope must also be destroyed. (plenty of docs on this, e.g. http://www.mattzeunert.com/2014/11/03/manually-removing-angular-directives.html)

This can be observed by having a cell editor that has a directive with a $destroy handler. It will not be called, thus will not perform any cleanup, resulting in leaks and other strange behaviour.

ag-grid should be adding and removing elements from the DOM similar to how ng-if does it:
https://github.com/angular/angular.js/blob/master/src/ng/directive/ngIf.js#L3

If a cell is repeatedly switched between view/edit, new scopes will continue to leak.
Perhaps there needs to be a callback in the edit/renderer to specify whether a new child scope should be created (and subsequently destroyed). Or return the scope itself to use for compilation and we could call $destroy on it ourselves in our destroy handler.

Possibly related to: https://github.com/ceolter/ag-grid/issues/1234

**Minimal reproduction of the problem with instructions**

An angularjs application with a cell editor template which uses a directive with a $destroy handler. The destroy handler will not be called when the element is removed.

**What is the motivation / use case for changing the behavior?**

No $destroy of child scopes results in memory leaks and bugs.

**Please tell us about your environment:**

* **ag-Grid version:** 11.0.0

* **Browser:**

Chrome

* **Language:** TypeScript

**Comment by smhc** [comment]
A plunkr which demonstrates the bug:
https://plnkr.co/edit/01wnYnEzTpf2WrfJCXpG?p=preview

Click on the cells a few times and leak Sausages like crazy.
Only when the row scopes are gone (ngif on the grid) will the destruction of the created scopes occur.

**Comment by smhc** [comment]
This workaround uses a transclude directive that introduces a new scope and marks it so it can be found easily. On ag-grid destroy we look through all the child scopes for the matching scope and destroy it.

https://plnkr.co/edit/fRONh1Tcmm1GnPq4pjM2?p=preview

Still a bit buggy, but the scope destroys do seem to be getting called.

**Comment by ceolter** [comment]
i fixed this a few months ago, i didn't realise there was an open issue for it.

the cells are now compiled after cell editors are put in. this is the line of code that does it:
https://github.com/ag-grid/ag-grid/blob/master/src/ts/rendering/cellComp.ts#L955

**Comment by phalgunv** [comment]
@ceolter Is the fix for this bug available in the latest version(v18.1.2) of ag-grid?
